pyocd/probe/pydapaccess/interface/pyusb_backend.py

```diff
@@ -169,9 +169,9 @@ def get_all_connected_interfaces():
             new_board = PyUSB()
             new_board.vid = board.idVendor
             new_board.pid = board.idProduct
-            new_board.product_name = board.product
-            new_board.vendor_name = board.manufacturer
-            new_board.serial_number = board.serial_number
+            new_board.product_name = board.product or ""
+            new_board.vendor_name = board.manufacturer or ""
+            new_board.serial_number = board.serial_number or ""
```

Comment:
I'd like to have some default serial number value here.

How about:

```py
base64.b32encode(
  struct.pack('Q', (
    hash(str(board.bus)) ^ hash(str(board.address))) ^ 
    hash(str(board.idVendor)) ^ hash(str(board.idProduct))
    )
  ).strip(b'=').decode('ascii')
```

This will generate a consistent serial number for a probe plugged into the same port and bus. For a random device I have plugged in, it generates "PWY3DCDKBUVTK".

Probably total overkill. 😅

Reply:
I have added this to pyusb backend, using SHA1. The builtin `hash` function is randomized on python startup.

Reply:
Much better than mine. 💯